Add offline install support

jwakre opened the pull request:
Fixes Issue: #234 

Changes proposed in this pull request:
- Add offline install support
- TUI: Update network check workflow for offline installs
- GUI: Update network check workflow for offline installs
- GUI: Add network check menu entry
- GUI: Create pop-up box for failing system checks
- GUI: Remove pre-check page

dorileo commented on controller/controller.go:
```diff
@@ -769,3 +778,30 @@ func generateISO(rootDir string, md *model.SystemInstall, options args.Args) err
+	customPath := path.Join(chrootPath, conf.CustomConfigDir)
+
+	info, err := os.Stat(customPath)
+	if os.IsNotExist(err) {
```
Why not use `utils.FileExists()` ? Also I don't think you need to check if it's directory since the environment is very controlled, we're pretty sure there won't be a directory in that path.

mdhorn commented on swupd/swupd.go:
```diff
@@ -378,6 +411,29 @@ func (s *SoftwareUpdater) VerifyWithBundles(version string, mirror string, bundl
+	}
+	defer func() { _ = os.RemoveAll(s.rootDir) }()
+
+	return s.VerifyWithBundles(version, mirror, bundles)
```
Now, if you download offline content and are generating an ISO, we call `VerifyWithBundles` three times -- the output is nearly identical each time and could be misleading to the user.

Please add a new parameter to  `VerifyWithBundles` with a prefix string for all of the steps. For the main OS install we can use a prefix like 'Target OS', for the offline content 'Offline Content', and for the ISO, 'ISO Boot OS'.

nazar2sfive commented:
what is the progress? 
without this we have to download content twice with 256KiB/s.

jwakre commented:
I made the following changes:
* Addressed review feedback
* Added documentation
* Improved offline install performance 
* Fixed indeterminate progress bar

jwakre commented on controller/controller.go:
```diff
@@ -581,6 +593,41 @@ func contentInstall(rootDir string, version string, md *model.SystemInstall, opt
 	return nil, nil
 }
 
+func copyOfflineToStatedir(rootDir, stateDir string) error {
```
This function doesn't handle the case when the statedir already exists.

jwakre commented on scripts/InstallerYAMLSyntax.md:
```diff
@@ -117,6 +123,7 @@ Item | Description | Default
 `hostname` | Name of the host system | `-UNIQUE RANDOM-`
 `version` | Version of Clear Linux OS to install | `-VERSION_ON_BUILD_SYSTEM-`
 `autoUpdate` | Should the system automatically update to the latest release of Clear Linux OS as part of the installation?; true or false | true
+`offline` | Install update content for minimal offline installation | false
```
I'm changing offline to default true.

jwakre commented on isoutils/isoutils.go:
```diff
@@ -112,6 +113,46 @@ func mkRootfs() error {
+	args := []string{
+		"losetup",
+		"-j",
+		"/mnt/media/images/roootfs.img",
```
Typo:

s/roootfs/rootfs

jwakre commented:
@mdhorn I found 3 more issues that I will push fixes for soon:

1. Misspelled `/mnt/media/images/rootfs.img` which prevented ISO from using unsquashfs
2. When toggling Gparted with advanced disk partitioning, the welcome page is loaded instead of the menu page
3. When a system check fails in the GUI, 2 dialog windows are created

jwakre commented:
I made the following changes:

1. Fixed typo (roootfs -> rootfs) which re-enables support for unsquashfs
2. Updated the welcome page logic to properly reload the clr-installer when switching from Gparted and to prevent 2 dialog pop-ups when the system check fails.
3. Increased the indeterminate progress bar speed.

mdhorn commented:
LGTM 👍

nazar2sfive commented:
version : clear-31090-live-desktop.iso

it fails to proceed if internet is not detected.
![image](https://user-images.githubusercontent.com/12841320/65832900-1b509000-e2ea-11e9-8a57-598e304eb02f.png)

@mdhorn from which version is offline install supported?

jwakre commented:
@nazar2sfive The installer images at https://clearlinux.org/downloads do not have offline install support yet. The clr-installer must make a release before images with offline installer support can be distributed which should happen soon. I will close #234 when installer images with offline support are published.

malventano commented:
Did this capability regress? Attempting a basic install of `clear-41190-live-server.iso` without network still attempts to connect to the network and fails. No packages were selected. Full defaults.
